server/store/memory/peer_store_test.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2016 The Chihaya Authors. All rights reserved.
+// Use of this source code is governed by the BSD 2-Clause license,
+// which can be found in the LICENSE file.
+
+package memory
+
+import (
+	"net"
+	"testing"
+
+	"github.com/chihaya/chihaya"
+	"github.com/chihaya/chihaya/server/store"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	hash = chihaya.InfoHash("11111111111111111111")
+
+	peers = []struct {
+		seeder bool
+		peerID string
+		ip     string
+		port   uint16
+	}{
+		{false, "-AZ3034-6wfG2wk6wWLc", "250.183.81.177", 5720},
+		{false, "-AZ3042-6ozMq5q6Q3NX", "38.241.13.19", 4833},
+		{false, "-BS5820-oy4La2MWGEFj", "fd45:7856:3dae::48", 2878},
+		{false, "-AR6360-6oZyyMWoOOBe", "fd0a:29a8:8445::38", 3167},
+		{true, "-AG2083-s1hiF8vGAAg0", "231.231.49.173", 1453},
+		{true, "-AG3003-lEl2Mm4NEO4n", "254.99.84.77", 7032},
+		{true, "-MR1100-00HS~T7*65rm", "211.229.100.129", 2614},
+		{true, "-LK0140-ATIV~nbEQAMr", "fdad:c435:bf79::12", 4114},
+		{true, "-KT2210-347143496631", "fdda:1b35:7d6e::9", 6179},
+		{true, "-TR0960-6ep6svaa61r4", "fd7f:78f0:4c77::55", 4727},
+	}
+	unmarshalledConfig = struct {
+		Shards int
+	}{
+		1,
+	}
+	config = store.DriverConfig{
+		"memory",
+		unmarshalledConfig,
+	}
+)
+
+func peerInSlice(peer chihaya.Peer, peers []chihaya.Peer) bool {
+	for _, v := range peers {
+		if v.Equal(&peer) {
+			return true
+		}
+	}
+	return false
+}
+
+func TestPeerStore(t *testing.T) {
```

Comment:
Can this test either be named better or broken into smaller tests that test one thing?